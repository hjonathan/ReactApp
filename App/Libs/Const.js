import RNFS from "react-native-fs";

export const CASE = {
    INBOX: "Inbox",
    DRAFT: "Draft",
    PARTICIPATED: "Participated",
    UNASSIGNED: "Unassigned",
    STATUS: {
        NONE: "none",
        SENDING: "sending",
        WORKING: "working",
        OFFLINE: "offline"
    }
};

export const DATETIME = {
    LOCAL: "YYYY-MM-DD HH:mm:ss",
    LOCAL_UTC: "YYYY-MM-DD HH:mm:ss Z"
};

export const DATE = {
    SIMPLE: "YYYYMMDD",
    LOCAL: "YYYY-MM-DD"
};

export const TIME = {
    LOCAL: "HH:mm:ss"
};

export const DS = "/";

export const PATH = {
    PROJECT: `${RNFS.DocumentDirectoryPath}`,
    FORMRENDER: `${DS}FormRender`,
    STEPS: `${DS}Steps`,
    NEXTSTEPS: `${DS}NextSteps`,
    FROMS: `${DS}Forms`,
    DATA: `${DS}Data`,
    EXTERNALLIBS: `${DS}ExternalLibs`,
    BUILDPROD: `${DS}build-prod`,
    MUSIC: "/storage/emulated/0/Music/processmakeraudiorecord/"
};

export const FILENAMES = {
    EXTERNALMAP: "ExtMap.json"
};
