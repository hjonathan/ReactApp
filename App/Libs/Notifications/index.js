/* eslint-disable import/prefer-default-export */
import * as pushNotification from "./pushNotification";

export {pushNotification};
