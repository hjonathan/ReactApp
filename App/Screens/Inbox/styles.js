import { StyleSheet } from 'react-native';
export default StyleSheet.create({
    mainContainer: {
        flex: 1,
        flexDirection: 'row',
        paddingVertical: 8,
        paddingHorizontal: 8,
        backgroundColor: '#2B6697',
    },
    icon: {
        width: 24,
        height: 24,
    }
});
