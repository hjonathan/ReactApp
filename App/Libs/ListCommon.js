import _ from "lodash";
import Task from "../Model/Task";
import Step from "../Model/Step";
import {CASE} from "./Const";

/**
 * Function commons for: inbox, draft, unassigned and participated
 */
export default class ListCommon {
    /**
     * Constructor
     * @param props
     */
    constructor (props) {
        this.props = props;
        this.task = Task.allTaskIdOffline();
    }

    /**
     * Get data lists
     * @param method
     * @param params
     */
    getDataList (method, params = {}) {
        if (this.props[method]) {
            this.props[method](params);
        }
    }

    /**
     * Gets usersInfo
     * @param caseList
     * @param type is type in list: prevUser, currentUser
     */
    getUsersInfo (caseList, type) {
        if (caseList.length > 0 && caseList[0][type]) {
            const ids = caseList.map(cases => cases[type].userId);
            this.props.userInfoRequest({
                size: "1",
                user: {
                    ids: _.uniq(ids)
                }
            });
        }
    }

    /**
     * Merges userPhoto props to dataList
     * @param params
     * @param type is type in list: prevUser, currentUser
     */
    mergeList (params, type) {
        let i,
            j,
            cbUser = function (user) { return user.userId === params.caseList[i][type].userId; };
        if (params.caseList) {
            for (i = 0; i < params.caseList.length; i += 1) {
                if (params.caseList[i][type]) {
                    j = _.findIndex(params.users, cbUser);
                    if (j >= 0) {
                        params.caseList[i].userPhoto = params.users[j].userPhoto;
                    }
                    params.caseList[i].status = this.processIfOffline(params.caseList[i]);
                }
            }
        }
        return params.caseList;
    }

    /**
     * Check if case is offline and get all necessary for case
     * @param caseData
     * @returns {boolean}
     */
    processIfOffline (caseData) {
        let isOffline,
            {status} = caseData;
        if (!status || status === CASE.STATUS.NONE) {
            if (caseData.task.offlineEnabled === "TRUE" && this.props.caseOffline) {
                this.props.caseOffline(caseData);
                status = CASE.STATUS.OFFLINE;
            }
            isOffline = Step.existSteps(caseData.process.processId, caseData.task.taskId);
            if (isOffline && _.includes(this.task, caseData.task.taskId)) {
                status = CASE.STATUS.OFFLINE;
            }
        }
        return status;
    }

    /**
     * Key for FlatList formed by delIndex and caseId
     * @param item
     * @returns {string}
     */
    keyExtractor (item, index) {
        return `${item.delIndex}_${item.caseId}`;
    }
}
