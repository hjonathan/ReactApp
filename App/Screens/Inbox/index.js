import {connect} from "react-redux";
import InboxScreen from "./inbox";
import actions from "../../Actions";
import {CASE} from "../../Libs/Const";

/**
 * Return state from Store to View Component
 * @param state
 */
const mapStateToProps = state => ({
        cases: {
            todo: state.Cases.todo,
            users: state.Cases.users
        },
        list: CASE.INBOX,
        forceRefresh: state.Cases.forceRefresh || false,
        disableItemList: state.Screens.All.disableItemList,
        isConnected: state.Net.isConnected
    }),
    mapDispatchToProps = dispatch => ({
        casesTodoRequest: params => dispatch(actions.cases.inbox.request(params)),
        userInfoRequest: params => dispatch(actions.cases.userInfo.request(params)),
        caseOffline: params => dispatch(actions.net.case.list(params)),
        routeName: name => dispatch(actions.cases.routeName(name)),
        reset: () => dispatch(actions.cases.reset())
    });

export default connect(mapStateToProps, mapDispatchToProps)(InboxScreen);
