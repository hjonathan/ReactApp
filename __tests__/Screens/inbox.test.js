import React from 'react'
import {shallow} from 'enzyme';
import Inbox from '../../App/Screens/Inbox';
import InboxScreen from '../../App/Screens/Inbox/inbox';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import {createStore} from 'redux';
import {CASE} from "../../App/Libs/Const";

jest.mock('react-native-orientation', () => require.requireActual('../../__mocks__/react-native-orientation').default);

describe('Screen >>> Inbox', () => {
    const initialState = {
            Cases: {
                todo: {
                    response: [
                        {
                            prevUser: {
                                userId: "00001"
                            }
                        },
                        {
                            prevUser: {
                                userId: "00002"
                            }
                        }
                    ]
                },
                users: [
                    {
                        userId: "00001",
                        userPhoto: "picture1"
                    },
                    {
                        userId: "00002",
                        userPhoto: "picture2"
                    }
                ]
            },
            Screens: {
                All: {
                    disabledItemList: false
                }
            },
            Net: {
                isConnected: true
            }
        },
        casesTodoRequest = jest.fn(),
        userInfoRequest = jest.fn(),
        caseOffline = jest.fn(),
        routeName = jest.fn(),
        reset = jest.fn(),
        navigation = {
            dispatch: jest.fn(),
            setParams: jest.fn(),
            state: {
                routeName: CASE.DRAFT,
                params:{
                    searchInput: null
                }
            }
        },
        mockStore = configureStore();

    let store,
        container,
        viewInbox,
        viewInboxOff;

    beforeEach(() => {
        store = mockStore(initialState);
        container = shallow(<Inbox store={store} />);
        viewInbox = shallow(
            <InboxScreen
                navigation={navigation}
                isConnected={true}
                casesTodoRequest={casesTodoRequest}
                userInfoRequest={userInfoRequest}
                caseOffline={caseOffline}
                routeName={routeName}
                reset={reset}
                store={store}
            />
        );
        viewInboxOff = shallow(<InboxScreen navigation={navigation} isConnected={false} routeName={routeName}/>);
    });

    /**
     * Render test
     */
    it('Render inbox correctly', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('mapStateToProps', () => {
        expect(container.props('children').casesTodoRequest()).toMatchSnapshot();
        expect(container.props('children').userInfoRequest()).toMatchSnapshot();
        expect(container.props('children').caseOffline()).toMatchSnapshot();
        expect(container.props('children').routeName()).toMatchSnapshot();
        expect(container.props('children').reset()).toMatchSnapshot();
    });

    it('Check static navigationOptions', () => {
        expect(InboxScreen.navigationOptions({navigation}).title)
            .toEqual("menu_drawer_main_menu_item_inbox");
        expect(InboxScreen.navigationOptions({navigation}).headerTintColor)
            .toEqual("#FFF");
        expect(InboxScreen.navigationOptions({navigation}).headerStyle)
            .toEqual({backgroundColor: "#1F6CBC"});
        expect(InboxScreen.navigationOptions({navigation}).headerLeft).toMatchSnapshot();
    });

    it("Should call navigationOptions method", () => {
        expect(viewInbox.instance().navigationOptions).toBeUndefined();
    });

    it("Props coverage cases", () => {
        expect(container.prop("cases")).toEqual(initialState.Cases);
    });

    it("Props coverage disableItemList", () => {
        expect(container.prop("disableItemList")).toEqual(initialState.Screens.All.disableItemList);
    });

    it("Props coverage componentWillReceiveProps", () => {
        viewInbox.instance().componentWillReceiveProps({
            cases: {
                todo: initialState.Cases.todo
            },
            forceRefresh: true,
            list: CASE.INBOX,
            disableItemList: initialState.Screens.All.disableItemList
        });
        expect(viewInbox.instance().state.todoRequested).toEqual(false);
        viewInbox.instance().componentWillReceiveProps({
            cases: {
                users: initialState.Cases.users
            },
            forceRefresh: true,
            list: CASE.INBOX,
            disableItemList: initialState.Screens.All.disableItemList
        });
        expect(viewInbox.instance().state).toEqual({
            data: [],
            todoRequested: false,
            searchInput: false,
            middleHeight: 617,
            usersRequested: false,
            loading: true
        });
        viewInbox.instance().setState({usersRequested: false});
        viewInbox.instance().componentWillReceiveProps({
            cases: {
                todo: {response: []},
                users: initialState.Cases.users
            },
            forceRefresh: true,
            list: CASE.INBOX,
            disableItemList: initialState.Screens.All.disableItemList
        });
        expect(viewInbox.instance().state.todoRequested).toEqual(false);
    });

    it("Should call onRefresh method", () => {
        expect(viewInbox.instance().onRefresh()).toMatchSnapshot();
        expect(viewInbox.instance().onRefresh({nativeEvent: {text: "abc"}})).toMatchSnapshot();
        expect(viewInboxOff.instance().onRefresh()).toMatchSnapshot();
    });

    it("Should call changeLayout method", () => {
        viewInbox.instance().changeLayout();
        expect(viewInbox.instance().state.middleHeight).toEqual(617);
    });

    it("Should call goToNewCases method", () => {
        expect(viewInbox.instance().goToNewCases()).toMatchSnapshot();
    });

    it("Should call refresh method of FlatList component", () => {
        const flatList = viewInbox.props("children").children[0];
        expect(flatList.props.renderItem({})).toMatchSnapshot();
        expect(flatList.props.onRefresh()).toMatchSnapshot();
    });

    it("Should call refresh method of FAB component", () => {
        const floatActionButton = viewInbox.props("children").children[1];
        expect(floatActionButton.props.onClickAction()).toMatchSnapshot();
    });
});
