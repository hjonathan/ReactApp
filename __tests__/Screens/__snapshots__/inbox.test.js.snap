// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen >>> Inbox Check static navigationOptions 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "padding": 16,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="bars"
    size={12}
    style={
      Object {
        "color": "#FFF",
        "fontSize": 20,
      }
    }
  />
</TouchableOpacity>
`;

exports[`Screen >>> Inbox Render inbox correctly 1`] = `
<Inbox
  caseOffline={[Function]}
  cases={
    Object {
      "todo": Object {
        "response": Array [
          Object {
            "prevUser": Object {
              "userId": "00001",
            },
          },
          Object {
            "prevUser": Object {
              "userId": "00002",
            },
          },
        ],
      },
      "users": Array [
        Object {
          "userId": "00001",
          "userPhoto": "picture1",
        },
        Object {
          "userId": "00002",
          "userPhoto": "picture2",
        },
      ],
    }
  }
  casesTodoRequest={[Function]}
  forceRefresh={false}
  isConnected={true}
  list="Inbox"
  reset={[Function]}
  routeName={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userInfoRequest={[Function]}
/>
`;

exports[`Screen >>> Inbox Should call goToNewCases method 1`] = `undefined`;

exports[`Screen >>> Inbox Should call onRefresh method 1`] = `undefined`;

exports[`Screen >>> Inbox Should call onRefresh method 2`] = `undefined`;

exports[`Screen >>> Inbox Should call onRefresh method 3`] = `undefined`;

exports[`Screen >>> Inbox Should call refresh method of FAB component 1`] = `undefined`;

exports[`Screen >>> Inbox Should call refresh method of FlatList component 1`] = `
<ItemList
  isConnected={true}
  parent={
    Object {
      "caseOffline": [MockFunction],
      "casesTodoRequest": [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {
              "search": "",
            },
          ],
          Array [
            Object {
              "search": "",
            },
          ],
          Array [
            Object {},
          ],
          Array [
            Object {
              "search": "",
            },
          ],
          Array [
            Object {
              "search": "abc",
            },
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "isConnected": true,
      "navigation": Object {
        "dispatch": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "routeName": "NewCases",
                "type": "Navigation/NAVIGATE",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "setParams": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "searchInput": false,
              },
            ],
            Array [
              Object {
                "searchInput": false,
              },
            ],
            Array [
              Object {
                "searchInput": false,
              },
            ],
            Array [
              Object {
                "searchInput": false,
              },
            ],
            Array [
              Object {
                "searchInput": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "state": Object {
          "params": Object {
            "searchInput": null,
          },
          "routeName": "Draft",
        },
      },
      "reset": [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "routeName": [MockFunction] {
        "calls": Array [
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
          Array [
            "Draft",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "userInfoRequest": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "size": "1",
              "user": Object {
                "ids": Array [
                  "00001",
                  "00002",
                ],
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
    }
  }
/>
`;

exports[`Screen >>> Inbox Should call refresh method of FlatList component 2`] = `undefined`;

exports[`Screen >>> Inbox mapStateToProps 1`] = `
Object {
  "type": "cases/inbox/request",
}
`;

exports[`Screen >>> Inbox mapStateToProps 2`] = `
Object {
  "type": "cases/userInfo/request",
}
`;

exports[`Screen >>> Inbox mapStateToProps 3`] = `
Object {
  "type": "net/case/list",
}
`;

exports[`Screen >>> Inbox mapStateToProps 4`] = `
Object {
  "type": "cases/routeName",
}
`;

exports[`Screen >>> Inbox mapStateToProps 5`] = `
Object {
  "type": "cases/reset",
}
`;
